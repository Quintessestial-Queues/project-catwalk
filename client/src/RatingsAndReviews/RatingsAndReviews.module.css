
/* div {
  border-style: solid;
} */

.ratingsAndReviews {
  display: grid;
  grid-template-columns: 1fr 2fr;
}