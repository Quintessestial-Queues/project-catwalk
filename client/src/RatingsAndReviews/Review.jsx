import React from 'react';
// import styles from './ReviewsList.module.css';


let Review = (props) => {
  return (
    <div className='review'>
      Review
      </div>
  )
}

export default Review;