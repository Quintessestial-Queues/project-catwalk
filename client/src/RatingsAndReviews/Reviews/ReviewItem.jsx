import React from 'react';
import StarRating from '../../SharedComponents/StarRating.jsx';
import moment from 'moment';
import { FaCheck } from 'react-icons/fa';
import { useState, useEffect, useRef } from 'react';
import PhotoModal from './PhotoModal.jsx';

import styles from './ReviewItem.module.css'

let ReviewItem = ({review}) => {
  const [helpfulnessCount, setCount] = useState(review.helpfulness);
  const [clicked, setClicked] = useState(false);
  const [showMore, toggleShowMore] = useState(false);
  const [showReviewModal, toggleReviewModal] = useState(false);

  const isFirstRender = useRef(true);

  let boldedSummary = review.summary.slice(0, 60);
  let restOfSummary = review.summary.length < 60 ? null : `...${review.summary.slice(60)}`;
  let body = !showMore ? review.body.slice(0, 250) : review.body;
  let showMoreButton = review.body.length > 250 && !showMore ? <button onClick={() => {
    toggleShowMore(!showMore);
  }}>Show More</button> : null;

  useEffect(() => {
    if (isFirstRender.current) {
      isFirstRender.current = false;
      return;
    }
    setClicked(true)
  }, [helpfulnessCount]);


  //On clicking of an image
    //Have a modal popup of the image in its native resolution

  //Ideas: Make a separate component for the images? That way you can just set a state and when that changes, the modal window should pop out

  return (
    <div className={styles.reviewContainer}>

      <span className={styles.starRating}><StarRating starRating={review.rating}/></span>
      <span className={styles.datePosted}>{`${review.reviewer_name} reviewed on ${moment(review.date).format("ll")}`}</span>

      <div className={styles.boldedSummary}>{boldedSummary} <p>{restOfSummary}</p></div>
      <p className={styles.body}>{body}{showMoreButton}</p>

      <div className={styles.photos}>
        {review.photos.length > 0 ? review.photos.map((photo, i) => {
          return (
            <PhotoModal key={i} photo={photo.url}/>
          )
        }) : null}
      </div>

      <span className={styles.recommendation}>{!review.recommend ? null : <span>
        <FaCheck/> {'I recommend this product'} </span>}</span>

      <div className={review.response ? styles.responseActive : styles.responseNormal}>{review.response ? `Response from seller: ${review.response}` : null}</div>

      <span className={styles.helpful}>Was this review helpful? <a onClick={() => {!clicked ? setCount(helpfulnessCount + 1) : console.log('Already clicked!')}}>Yes ({helpfulnessCount})</a>
      <a onClick={ () => {!clicked ? setCount(helpfulnessCount - 1) : console.log('Already clicked!')}}> No</a></span>
    </div>
  )
}

export default ReviewItem;
