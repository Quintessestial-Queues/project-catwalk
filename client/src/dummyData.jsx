
let dummyProducts = [
  {
      "id": 17067,
      "campus": "hr-rfp",
      "name": "Camo Onesie",
      "slogan": "Blend in to your crowd",
      "description": "The So Fatigues will wake you up and fit you in. This high energy camo will have you blending in to even the wildest surroundings.",
      "category": "Jackets",
      "default_price": "140.00",
      "created_at": "2021-02-23T04:22:44.728Z",
      "updated_at": "2021-02-23T04:22:44.728Z"
  },
  {
      "id": 17068,
      "campus": "hr-rfp",
      "name": "Bright Future Sunglasses",
      "slogan": "You've got to wear shades",
      "description": "Where you're going you might not need roads, but you definitely need some shades. Give those baby blues a rest and let the future shine bright on these timeless lenses.",
      "category": "Accessories",
      "default_price": "69.00",
      "created_at": "2021-02-23T04:22:44.728Z",
      "updated_at": "2021-02-23T04:22:44.728Z"
  },
  {
      "id": 17069,
      "campus": "hr-rfp",
      "name": "Morning Joggers",
      "slogan": "Make yourself a morning person",
      "description": "Whether you're a morning person or not.  Whether you're gym bound or not.  Everyone looks good in joggers.",
      "category": "Pants",
      "default_price": "40.00",
      "created_at": "2021-02-23T04:22:44.728Z",
      "updated_at": "2021-02-23T04:22:44.728Z"
  },
  {
      "id": 17070,
      "campus": "hr-rfp",
      "name": "Slacker's Slacks",
      "slogan": "Comfortable for everything, or nothing",
      "description": "I'll tell you how great they are after I nap for a bit.",
      "category": "Pants",
      "default_price": "65.00",
      "created_at": "2021-02-23T04:22:44.728Z",
      "updated_at": "2021-02-23T04:22:44.728Z"
  },
  {
      "id": 17071,
      "campus": "hr-rfp",
      "name": "Heir Force Ones",
      "slogan": "A sneaker dynasty",
      "description": "Now where da boxes where I keep mine? You should peep mine, maybe once or twice but never three times. I'm just a sneaker pro, I love Pumas and shell toes, but can't nothin compare to a fresh crispy white pearl",
      "category": "Kicks",
      "default_price": "99.00",
      "created_at": "2021-02-23T04:22:44.728Z",
      "updated_at": "2021-02-23T04:22:44.728Z"
  }
]


let dummyRelatedProducts = [
  17069,
  17073,
  17072,
  17071
]



let dummyProduct = {
"id": 17068,
"campus": "hr-rfp",
"name": "Bright Future Sunglasses",
"slogan": "You've got to wear shades",
"description": "Where you're going you might not need roads, but you definitely need some shades. Give those baby blues a rest and let the future shine bright on these timeless lenses.",
"category": "Accessories",
"default_price": "69.00",
"created_at": "2021-02-23T04:22:44.728Z",
"updated_at": "2021-02-23T04:22:44.728Z",
"features": [
    {
        "feature": "Lenses",
        "value": "Ultrasheen"
    },
    {
        "feature": "UV Protection",
        "value": null
    },
    {
        "feature": "Frames",
        "value": "LightCompose"
    }
]
}

let dummyReviews = {
"product": "17068",
"page": 0,
"count": 5,
"results": [
    {
        "review_id": 188758,
        "rating": 3,
        "summary": "I'm enjoying wearing these shades",
        "recommend": true,
        "response": "",
        "body": "Comfortable and practical.",
        "date": "2019-04-14T00:00:00.000Z",
        "reviewer_name": "shortandsweeet",
        "helpfulness": 8,
        "photos": [
            {
                "id": 298296,
                "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
            },
            {
                "id": 298297,
                "url": "https://images.unsplash.com/photo-1561693532-9ff59442a7db?ixlib=rb-1.2.1&auto=format&fit=crop&w=975&q=80"
            },
            {
                "id": 298298,
                "url": "https://images.unsplash.com/photo-1487349384428-12b47aca925e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80"
            }
        ]
    },
    {
        "review_id": 288940,
        "rating": 5,
        "summary": "Best Review Ever!",
        "recommend": true,
        "response": null,
        "body": "This Review is great, so is the person who made this module!",
        "date": "2021-03-12T00:00:00.000Z",
        "reviewer_name": "Tear-in",
        "helpfulness": 2,
        "photos": []
    },
    {
        "review_id": 288942,
        "rating": 5,
        "summary": "Cutest Frog",
        "recommend": true,
        "response": null,
        "body": "EVERRRRRRR!RRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRR",
        "date": "2021-03-12T00:00:00.000Z",
        "reviewer_name": "frogeeeeeee",
        "helpfulness": 0,
        "photos": []
    },
    {
        "review_id": 188760,
        "rating": 2,
        "summary": "This product was ok!",
        "recommend": false,
        "response": "",
        "body": "They're fine but I wouldn't buy again.",
        "date": "2019-05-23T00:00:00.000Z",
        "reviewer_name": "anyone",
        "helpfulness": 0,
        "photos": []
    },
    {
        "review_id": 188759,
        "rating": 5,
        "summary": "I'm not a fan!",
        "recommend": false,
        "response": "Sorry to hear. Is there anything in particular you don't like?",
        "body": "I don't like them",
        "date": "2019-06-16T00:00:00.000Z",
        "reviewer_name": "negativity",
        "helpfulness": 0,
        "photos": []
    }
]
}



export {dummyProducts, dummyProduct, dummyRelatedProducts, dummyReviews};