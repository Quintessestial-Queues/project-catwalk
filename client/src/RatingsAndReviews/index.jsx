import React from 'react';
import dummyReviews from '../dummyReviews.jsx';
import Review from './Review.jsx';
import SortFilter from './SortFilter.jsx';

import styles from './ReviewsList.module.css';

class ReviewsList extends React.Component {
  constructor(props) {
    super(props);

  }

  render() {
    //map over reviews from props
    let reviewItems = dummyReviews.map((review) => {
    return <Review reviewItem={review}/>
  })
  return (
    <div className='reviews-list'>
      <SortFilter />
      {reviewItems}
    </div>
  )
  }
}



export default ReviewsList;

