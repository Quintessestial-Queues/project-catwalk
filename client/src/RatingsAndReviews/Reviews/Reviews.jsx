import React, { useState, useContext, useEffect } from 'react';
import ReviewItem from './ReviewItem.jsx';
import CreateReview from './CreateReview.jsx';
import { RatingsAndReviewsContext } from '../../state/RatingsAndReviewsContext.js';
import { ProductContext } from '../../state/ProductContext.js';
import { APIContext } from '../../state/APIContext.js';

import styles from './Reviews.module.css';

const Reviews = (props) => {
  const { reviews, filteredReviews, setReviews } = useContext(RatingsAndReviewsContext);
  const { getReviews } = useContext(APIContext);
  const { productId } = useContext(ProductContext);

  const [loading, setLoading] = useState(false);
  const [clickedMoreReviews, setClickedMoreReviews] = useState(false);
  const [hasMore, setHasMore] = useState(false);
  const [showCreateReviewModal, setShowCreateReviewModal] = useState(false);

  const [reviewsView, setReviewsView] = useState(2);
  const [sortDropdown, setSortDropdown] = useState('relevance');

  useEffect(() => {
    getReviews(productId)
      .then(({data}) => {
        setReviews(data.results)
      })
      .catch((err) => {
        console.log('Error getting reviews in Reviews Component', err);
      })
  }, [productId])

  const handleScroll = (event) => {
    // TODO:implement me
  }

  const handleOnChangeSort = (e) => {
    getReviews(productId, e.target.value)
      .then(({ data }) => {
        setReviews(data.results);
      });
  }

  const handleClickMoreReviews = (event) => {
    setReviewsView(reviewsView + 2);
  }

  return (
    <div className={styles.reviewsContainer} onScroll={handleScroll}>
      <div className={styles.reviewSorter}>
        <p>
          {`${filteredReviews.length} reviews, sorted by `}
          <select onChange={handleOnChangeSort}>
            <option value='relevance'>Relevance</option>
            <option value='helpful'>Helpful</option>
            <option value='newest'>Newest</option>
          </select>
        </p>
      </div>
      <div className={styles.reviewsList}>
        {filteredReviews.slice(0, reviewsView).map((review, i) =>
          <ReviewItem key={i} review={review}/>
        )}
      </div>
      <div className='buttons'>
        {reviewsView < filteredReviews.length ? <button className={styles.moreReviewsButton} onClick={handleClickMoreReviews}>More Reviews</button> : null}
        <CreateReview showCreateReviewModal={showCreateReviewModal} />
      </div>
    </div>
  );
}


export default Reviews;
