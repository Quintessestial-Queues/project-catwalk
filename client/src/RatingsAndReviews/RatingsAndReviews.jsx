import React, { useContext, useEffect } from 'react';
import Ratings from './Ratings/Ratings.jsx'
import Reviews from './Reviews/Reviews.jsx';
import { dummyReviews } from '../dummyData.js';

import { APIContext } from '../state/APIContext';
import { RatingsAndReviewsContext } from '../state/RatingsAndReviewsContext.js';
import { ProductContext } from '../state/ProductContext.js';

import styles from './RatingsAndReviews.module.css';

const RatingsAndReviews = () => {

  const testContext = useContext(RatingsAndReviewsContext);
  const { getReviews } = useContext(APIContext);
  const { productId } = useContext(ProductContext);

  useEffect(() => {
    getReviews(productId)
      .then(({data}) => {
        console.log(data);
        // debugger;
        testContext.setReviews(data.results);
      })
      .catch((err) => {
        console.log('Error getting reviews', err);
      })
  }, [])

    return (
    <div className={styles.ratingsAndReviews}>
      <div><Ratings /></div>
      <div><Reviews /></div>
      </div>
    )

}

export default RatingsAndReviews;

