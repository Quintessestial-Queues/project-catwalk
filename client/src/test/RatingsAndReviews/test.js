import React from 'react';
// import chai from 'chai'
import { dummyProduct, dummyProductStyles, dummyReviews } from '../../dummyData';
import RatingsAndReviews from '../../RatingsAndReviews/RatingsAndReviews.jsx';
import Reviews from '../../RatingsAndReviews/Reviews/Reviews.jsx';
import ReviewItem from '../../RatingsAndReviews/Reviews/ReviewItem.jsx';

// import { should } from 'chai';
import { shallow, mount, render } from 'enzyme';


describe('RatingsAndReviews', () => {

  it ('renders the RatingsAndReviews component', () => {
    const wrapper = shallow(<RatingsAndReviews />);
    expect(wrapper).toMatchSnapshot();
  })


  describe('Reviews', () => {
    it ('renders two reviews by default when there are two or more  reviews', () => {
      const wrapper = shallow(<Reviews reviews={dummyReviews.results}/>);
      expect(wrapper.find(ReviewItem)).toHaveLength(2);
    })


    //TODOS: Refactor using a mock callback function to test click event instead
    it ('renders four reviews when button the "More Reviews" button is clicked', () => {
      const wrapper = shallow(<Reviews reviews={dummyReviews.results}/>);
      wrapper.setState({
        clickedMoreReviews: true,
        reviewsView: 4
      });
      expect(wrapper.find(ReviewItem)).toHaveLength(4);
    })
  })

});