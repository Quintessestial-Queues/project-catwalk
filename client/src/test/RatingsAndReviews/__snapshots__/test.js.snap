// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsAndReviews renders the RatingsAndReviews component 1`] = `
<div
  className="ratingsAndReviews"
>
  <div>
    <Ratings />
  </div>
  <div>
    <Reviews
      reviews={
        Array [
          Object {
            "body": "Comfortable and practical.",
            "date": "2019-04-14T00:00:00.000Z",
            "helpfulness": 8,
            "photos": Array [
              Object {
                "id": 298296,
                "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80",
              },
              Object {
                "id": 298297,
                "url": "https://images.unsplash.com/photo-1561693532-9ff59442a7db?ixlib=rb-1.2.1&auto=format&fit=crop&w=975&q=80",
              },
              Object {
                "id": 298298,
                "url": "https://images.unsplash.com/photo-1487349384428-12b47aca925e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1650&q=80",
              },
            ],
            "rating": 3,
            "recommend": true,
            "response": "",
            "review_id": 188758,
            "reviewer_name": "shortandsweeet",
            "summary": "I'm enjoying wearing these shades",
          },
          Object {
            "body": "This Review is great, so is the person who made this module!",
            "date": "2021-03-12T00:00:00.000Z",
            "helpfulness": 2,
            "photos": Array [],
            "rating": 5,
            "recommend": true,
            "response": null,
            "review_id": 288940,
            "reviewer_name": "Tear-in",
            "summary": "Best Review Ever!",
          },
          Object {
            "body": "EVERRRRRRR!RRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRR",
            "date": "2021-03-12T00:00:00.000Z",
            "helpfulness": 0,
            "photos": Array [],
            "rating": 5,
            "recommend": true,
            "response": null,
            "review_id": 288942,
            "reviewer_name": "frogeeeeeee",
            "summary": "Cutest Frog",
          },
          Object {
            "body": "They're fine but I wouldn't buy again.",
            "date": "2019-05-23T00:00:00.000Z",
            "helpfulness": 0,
            "photos": Array [],
            "rating": 2,
            "recommend": false,
            "response": "",
            "review_id": 188760,
            "reviewer_name": "anyone",
            "summary": "This product was ok!",
          },
          Object {
            "body": "I don't like them",
            "date": "2019-06-16T00:00:00.000Z",
            "helpfulness": 0,
            "photos": Array [],
            "rating": 5,
            "recommend": false,
            "response": "Sorry to hear. Is there anything in particular you don't like?",
            "review_id": 188759,
            "reviewer_name": "negativity",
            "summary": "I'm not a fan!",
          },
        ]
      }
    />
  </div>
</div>
`;
